web/src/hooks/useTeams.ts

```diff
@@ -28,3 +37,106 @@ interface createNewTeamForNamespaceParams {
   name: string;
   description: string;
 }
+
+export interface ITeams {
+  name: string;
+  description: string;
+  role: string;
+  avatar: IAvatar;
+  can_view: boolean;
+  repo_count: number;
+  member_count: number;
+  is_synced: boolean;
+}
+
+export function useFetchTeams(orgName: string) {
+  const [page, setPage] = useState(1);
+  const [perPage, setPerPage] = useState(10);
+  const [search, setSearch] = useState<SearchState>({
+    query: '',
+    field: teamViewColumnNames.teamName,
+  });
+
+  const {
+    data,
+    isLoading,
+    isPlaceholderData,
+    isError: errorLoadingTeams,
+  } = useQuery<ITeams[]>(
+    ['teams'],
+    ({signal}) => fetchTeamsForNamespace(orgName, signal),
+    {
+      placeholderData: [],
+    },
+  );
+
+  const teams: ITeams[] = Object.values(data);
+
+  const filteredTeams =
+    search.query !== ''
+      ? teams?.filter((team) => team.name.includes(search.query))
+      : teams;
+
+  const paginatedTeams = filteredTeams?.slice(
+    page * perPage - perPage,
+    page * perPage - perPage + perPage,
+  );
+
+  return {
+    teams: teams,
+    paginatedTeams: paginatedTeams,
+    loading: isLoading || isPlaceholderData,
+    error: errorLoadingTeams,
+    page,
+    setPage,
+    perPage,
+    setPerPage,
+    search,
+    setSearch,
+  };
+}
+
+export function useDeleteTeam({orgName, onSuccess, onError}) {
```

Comment:
I like how each hook is separated within the same file. We should keep this pattern.